lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -835,48 +628,56 @@ def make_mpdboot_options(self):
         else:
             self.log.debug('No mpdboot interface option')
 
+        # add the number of mpi processes (aka mpi universe) to mpdboot options
         if self.options.universe is not None and self.options.universe > 0:
             self.mpdboot_options.append("--ncpus=%s" % self.get_universe_ncpus())
 
-        # number of mpi
+        # if mpich, add the number of unique nodes as mpdboot_totalnum
         if self.mpdboot_totalnum:
             self.mpdboot_options.append("--totalnum=%s" % self.mpdboot_totalnum)
 
-        # verbosity
+        # set verbosity
         if self.options.mpdbootverbose:
             self.mpdboot_options.append("--verbose")
 
         # mpdboot rsh command
         if not self.has_hydra:
             self.mpdboot_options.append(self.MPDBOOT_TEMPLATE_REMOTE_OPTION_NAME % {'rsh': self.get_rsh()})
 
-    def mpdboot_set_localhost_interface(self):
+    ### BEGIN mpiexec ###
+    def mpiexec_set_global_options(self):
+        """Set mpiexec global options
+
+        unless explicitly asked not to, will add all environment variables to mpiexec_global_options
         """
-        Set the localhost mpdboot interface
+        self.mpiexec_global_options['MKL_NUM_THREADS'] = '1'
+
+        if not self.options.noenvmodules:
+            for env_var in self.GLOBAL_VARIABLES_ENVIRONMENT_MODULES:
+                if env_var in os.environ and env_var not in self.mpiexec_global_options:
+                    self.mpiexec_global_options[env_var] = os.environ[env_var]
+
+    def mpiexec_set_local_options(self):
```

Comment:
Function definitions only require an indented block and apparently docstrings count. I've left it because I didn't know for sure if it had a purpose, should I remove it?

This hasn't been changed in the PR, only moved
https://github.com/hpcugent/vsc-mympirun/blob/master/lib/vsc/mympirun/mpi/mpi.py#L990